tests/testthat/test-carehomes-check.R

```diff
@@ -0,0 +1,616 @@
+context("carehomes (check)")
+
+test_that("N_tot and N_tot2 stay constant", {
+  p <- carehomes_parameters(0, "uk")
+  mod <- carehomes$new(p, 0, 1)
+  info <- mod$info()
+  y0 <- carehomes_initial(info, 1, p)$state
+  mod$set_state(carehomes_initial(info, 1, p)$state)
+  mod$set_index(integer(0))
+  y <- mod$transform_variables(
+    drop(dust::dust_simulate(mod, seq(0, 400, by = 4))))
+
+  skip("This needs checking")
+
+  ## This is not quite correct, and I don't really know why. I think
```

Comment:
I've tracked this back and you're right, the discrepancy occurs in 'carehomes_population' due to the rounding when removing residents/ workers from their respective groups. Maybe we should write a function in utils that fixes this? eg

```suggestion
round_int <- function(x) {
 x_floor <- floor(x)
 n_round_up <- n - sum(x_floor)
 i <- order(x - x_floor, decreasing = TRUE)[seq_len(n_round_up)]
 x_floor[i] <- x_floor[i] + 1
 x_floor
}
```